### PR TITLE
Fix for WFCORE-1843. module command to export dependencies.

### PR DESCRIPTION
**Pull request by jfdenise**
New option for the module command and associated unit tests.

### CODE REVIEW

File: cli/src/main/resources/help/module.txt [hunk 1]
```diff
@@ -85,6 +86,14 @@ ARGUMENTS
                    is generated, i.e. when the --module-xml argument isn't
                    specified.
 
+ --export-dependencies  - (used with add, optional) a comma-separated list of module
+                   names the module being added depends on and are exported.
```
Comment (bstansberry) [inline cli/src/main/resources/help/module.txt]: I would say

"names the module being added depends on and exports."

File: cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java [hunk 2]
```diff
@@ -396,13 +416,30 @@ protected void addModule(CommandContext ctx, final ParsedCommandLine parsedCmd)
...
-        if(config != null) {
+        if (config != null) {
+            Set<String> modules = new HashSet<>();
             final String dependenciesStr = dependencies.getValue(parsedCmd);
             if(dependenciesStr != null) {
                 final String[] depsArr = dependenciesStr.split(",+");
                 for(String dep : depsArr) {
                     // TODO validate dependencies
                     config.addDependency(new ModuleDependency(dep));
+                    modules.add(dep);
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java]: Should this have the same kind of duplication check you do at L436 below?

**Comment by bstansberry** [comment]
Couple minor points aside, this looks good. But I'll delay merging it until QE is done with their analysis step. Please ping me when that is done.

File: cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java (hunk 2 above)
Comment (jfdenise) [inline cli/src/main/java/org/jboss/as/cli/handlers/module/ASModuleHandler.java]: I didn't wanted to change the existing behavior. We could have customer scripts that wrongly add dependencies twice.

**Comment by jfdenise** [comment]
@bstansberry, QE is done.
JF

**Pull request merged**